Remove vestigial Windows support and defunct Irix support

jonjensen opened the pull request:
The documentation has said for 15+ years:

"Microsoft Windows is not (and will not be) supported."

https://www.interchangecommerce.org/docs/install.html

msjohns1 reviewed (commented):
Generally fine. Review 2 in-line comments.

jonjensen commented on extensions/Interchange.pm:
```diff
@@ -445,11 +444,9 @@ sub get_filename {
 sub file_name_is_absolute {
     my($file) = @_;
-    $file =~ m{$abspat}oi ;
+    $file =~ m{^/}oi;
```
Good catch! Doesn't need the `i` flag anymore either.

jonjensen commented on lib/Vend/File.pm:
```diff
@@ -517,7 +511,7 @@ sub get_filename {
 
-my $abspat = $^O =~ /win32/i ? qr{^([a-zA-Z]:)?[\\/]} : qr{^/};
+my $abspat = qr{^/};
 my $relpat = qr{\.\.[\\/]};
```
Yep, another good catch. Removed.

jonjensen commented:
@msjohns1 I noticed some other obvious old Windows junk right there and removed it. But that was more invasive so I added some testing to the unit test catalog. Please see the new commits.

jonjensen commented:
@msjohns1 In addition to the earlier commits added here, today I found and removed more cruft, so please see more new commits here. Thanks.

msjohns1 reviewed (approved):
No issues.

jonjensen commented:
@msjohns1 Thanks for the review.
